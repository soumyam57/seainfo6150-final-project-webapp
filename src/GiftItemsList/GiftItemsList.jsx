import React from "react";
import PropTypes from "prop-types";
import GiftItem from "../GiftItem/GiftItem.jsx";
import styles from "./GiftItemsList.module.css";
import { Link } from "react-router-dom";

const GiftItemsList = props => {

  return (
    <div className={styles.flexContainer}>
      <h1 className={styles.title}>{props.pagetitle}</h1>
      <ul className={styles.container}>
            {props.giftItemList.map((giftItem,index) => (
                <li key={index}>
                  <Link  key={index} className={styles.link} to={'/'+giftItem.category+'/'+giftItem.categoryid}>
                    <GiftItem giftItem={giftItem}/>
                  </Link>
                </li>
            ))}
      </ul>
    </div>
  );
};

GiftItemsList.propTypes = {
  giftItemList: PropTypes.array.isRequired,
  pagetitle: PropTypes.string.isRequired
};

export default GiftItemsList;