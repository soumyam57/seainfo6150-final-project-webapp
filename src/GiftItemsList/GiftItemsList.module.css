.container {
    background-color: rgb(83, 15, 15);
    box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap:20px;
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 1%;
    align-content: center;
    height: 100%;
  }

  
  @media screen and (min-width: 100px) {
    .container {
      grid-template-columns: repeat(1, 1fr);
    }
  }
  
  @media screen and (min-width: 700px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
    }
  }
  
  .link{
      text-decoration-line: none;
  }

  .div{ 
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
}

.title {
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5); 
    padding:2%;
    text-align: center;
    font-style: italic;
    font-weight: bold;
    text-transform: uppercase;
    color: gainsboro;
}

  
  