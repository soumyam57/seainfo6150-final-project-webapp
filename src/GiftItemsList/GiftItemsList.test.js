import React from "react";
import { render } from "@testing-library/react";
import GiftItemsList from "./GiftItemsList.jsx";
import { BrowserRouter } from 'react-router-dom';

const giftItem={
   "couplerings":{
      "price":"$1000",
      "title":"Couple Rings",
      "category":"wedding",
      "categoryid":"couplerings",
      "img":
      {
         "url":"https://images-na.ssl-images-amazon.com/images/I/51mfwq0NDEL._AC_UX522_.jpg",
         "height":400,
         "width":500
      }
   }
}

it("renders the home component correctly", () => {
  const { container } = render(
    <BrowserRouter>
        <GiftItemsList giftItemList={Object.values(giftItem)} pagetitle="Wedding & Party"/>
    </BrowserRouter>
    );
  expect(container).toMatchSnapshot();
});
