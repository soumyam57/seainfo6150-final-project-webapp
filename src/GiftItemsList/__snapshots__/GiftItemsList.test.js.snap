// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div
    class="flexContainer"
  >
    <h1
      class="title"
    >
      Wedding & Party
    </h1>
    <ul
      class="container"
    >
      <li>
        <a
          class="link"
          href="/wedding/couplerings"
        >
          <div
            class="giftItem"
          >
            <p
              class="text"
            >
              Couple Rings
            </p>
            <div>
              <img
                alt=""
                class="image"
                src="https://images-na.ssl-images-amazon.com/images/I/51mfwq0NDEL._AC_UX522_.jpg"
              />
            </div>
            <p
              class="text"
            >
              Price: 
              $1000
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
